### DISCUSSION TITLE
Height issue of pinned tabs

### DISCUSSION THREAD

**Discussion by c02y** [piroor/treestyletab#3236] [Q&A] [answered]
This is my snippet in `Advanced` section of TST settings
```
.tab {
  height: 20px;
}

.tab.pinned {
  height: 20px;
}
```
But pinned tabs and unpinned tabs are different

![image](https://user-images.githubusercontent.com/2135996/197317210-374b8254-4486-475a-9821-fcfa4442526b.png)

And 
```
.tab {
  height: 20px;
}

.tab.pinned {
  height: 10px;
}
```
![image](https://user-images.githubusercontent.com/2135996/197317240-48faa233-efcf-4851-aea8-9be2d5a4d2e8.png)

So how to set height of the pinned tabs?

**Comment by irvinm** [comment, marked as answer]
This might help:  https://github.com/piroor/treestyletab/issues/2725

**Reply by c02y** [reply]
I searched all issues, including the one you mentioned before opening this discussion. None of them works. 

Could you please be more specific?

**Reply by irvinm** [reply]
Ok, how about this one?  https://github.com/piroor/treestyletab/discussions/3213

Yes, I am avoiding re-researching this topic. :)

**Reply by irvinm** [reply]
Also, when you changed faviconizedTabScale (and reloaded the sidebar), does this not effectively do what you want?

![image](https://user-images.githubusercontent.com/979729/197360984-c19f4a99-ac09-4578-ae51-a3a35eded699.png)

![1.75 scaling](https://user-images.githubusercontent.com/979729/197361064-148839b8-59e0-4a13-94cf-9d75ebb1cb2b.png)
1.75 scaling

![1.00 scaling](https://user-images.githubusercontent.com/979729/197361076-c768e96e-7ffb-43b4-a74f-74883eaa3b82.png)
1.00 scaling

**Reply by c02y** [reply]
I think you misunderstood what I'm asking from the first beginning, I'm not using faviconized style of pinned tabs, I'm using list style of pinned tabs like unpinned tabs.

**Reply by irvinm** [reply]
Ah!  You are right ... will have to check it out.

**Reply by irvinm** [reply]
Does this do what you want?

```
tab-item.pinned {
  height: 10px;
}
```

**Reply by c02y** [reply]
Unfortunately, the effect is the same as `.tab.pinned {}`, I might have already tried that too before opening this discussion.

Does it work for you?

**Reply by irvinm** [reply]
Well, works as I would expect for my formatting ... (pinned tabs disabling faviconized style)

Default:
![image](https://user-images.githubusercontent.com/979729/197368313-122144b8-deee-4bbf-96fe-458666d6a268.png)

Height 10px:
![image](https://user-images.githubusercontent.com/979729/197368318-387e6bbe-6549-4db3-86b8-6b28de926c1d.png)

Obviously other things would need to be adjusted ...

**Reply by irvinm** [reply]
If you are referring to the extra space between the pinned tabs, that is because pinned tabs have Javascript that calculates the total size of the pinned tab area and then assigns a value to each pinned tab for the property "--pinned-position-top:".  I am not aware of a way via CSS to change that so that each tab is spaced a certain amount apart.  (Normal tabs are a list and you can influence things like padding and spacing between then easily)

I loaded a clean profile and reduced the tab size to 10px and see:
![image](https://user-images.githubusercontent.com/979729/197369072-cfe1c69d-dceb-4f40-bd8f-6e2bdc5754f9.png)

I then manually changed that property for the first 3 tabs ... and it looks like this:
![image](https://user-images.githubusercontent.com/979729/197369056-d513b079-6cc7-40c8-8f70-6ac724d9a0aa.png)

I don't know a way to change the pinned tab height ... *AND* the spacing between pinned tab *AND* the pinned tab area via CSS as it depends on the tab and what item it is and the total number of tabs.  (At least beyond my CSS knowledge if possible at all)

**Reply by piroor** [reply]
I've tried:

```css
tab-item {
  height: 16px;
  --tab-size: 16px;
}
```

and I got:
![image](https://user-images.githubusercontent.com/70062/197405417-01014833-ca15-4a9a-ba68-2154fa1d7d03.png)

Please note that you need to close-and-reopen the sidebar to apply new height of tabs, because TST calculates various sizes when the sidebar panel is loaded and the calculation result are not updated dynamically.

**Reply by c02y** [reply]
```
.tab {
  height: 20px;
}

.tab.pinned {
  height: 20px;
}
```
Actually works as expected I only have to reopen the sidebar as you said (note that .tab and .tab.pinned blocks are both needed, only .tab.pinned doesn't work for pinned tab)

And 
```
tab-item {
  height: 16px;
  --tab-size: 16px;
}
```
works for both pinned tabs and unpinned tabs. 

Thanks.

FYI: Sorry, Github won't let me accept the right comment in the middle as the solution, so I made the first reply the answer.

**Reply by irvinm** [reply]
@c02y you shouldn't need your first block of CSS at all.  The "tab-item" CSS covers both normal tabs and pinned tabs.  

@piroor thank you for the missing piece.  I should have been able to find the "--tab-size: 16px;" solution.  🙌 

![image](https://user-images.githubusercontent.com/979729/197411635-7eb3435b-dc05-4fa5-9885-f67683d87ab7.png)

**Reply by c02y** [reply]
> @c02y you shouldn't need your first block of CSS at all. The "tab-item" CSS covers both normal tabs and pinned tabs.
> 
> @piroor thank you for the missing piece. I should have been able to find the "--tab-size: 16px;" solution. raised_hands
> 
> ![image](https://user-images.githubusercontent.com/979729/197411635-7eb3435b-dc05-4fa5-9885-f67683d87ab7.png)

Yes, if I use `tab-item {}`, I don't need `.tab {}`, but since `tab-item {}` works for both pinned tabs and unpinned tabs, if I need to set different values for pinned and unpinned tabs, then use `.tab {}` and `.tab.pinned {}` together.

**Reply by irvinm** [reply]
@c02y ... gotcha!  Glad you got it to work.
